### PR TITLE
Add support for changing the repo owner via options

### PR DESCRIPTION
**Pull request by MichaelSimons**
Looking for feedback on the new option.  I can imagine a scenario in which specifying an entirely different repo may be useful.  That would require the mapping of old repo to new repo since the manifest can contain multiple repos.  The repo owner functionality I added satisfies the current requirements from the dotnet core repos.  Having to specify a map makes it clunky for this scenario which is why I didn't add it.  Adding another option down the road to specify a completely different repo is certainly possible.

### CODE REVIEW

File: src/Microsoft.DotNet.ImageBuilder/ImageBuilder.cs [hunk 1]
```diff
@@ -70,54 +72,26 @@ private static void Build()
...
+                finally
+                {
+                    if (createdPrivateDockerfile)
                     {
-                        filename = command.Substring(0, firstSpaceIndex);
-                        args = command.Substring(firstSpaceIndex + 1);
+                        File.Delete(dockerfilePath);
                     }
-
-                    ExecuteHelper.Execute(filename, args, Options.IsDryRun);
```
Comment (dagood) [inline src/Microsoft.DotNet.ImageBuilder/ImageBuilder.cs]: Are all the changes in the file for this PR? There are some things here that look like unrelated improvements.

Comment (MichaelSimons) [inline src/Microsoft.DotNet.ImageBuilder/ImageBuilder.cs]: Apologies, there were some methods that were not ordered correctly so I reordered them. The BuildImages method was getting too busy to I split it into separate methods but I consider those to be related.

File: src/Microsoft.DotNet.ImageBuilder/ImageBuilder.cs [hunk 2]
```diff
@@ -191,6 +181,72 @@ private static void PushImages()
...
+            dockerfilePath = Path.Combine(image.ActivePlatform.Model.Dockerfile, "Dockerfile");
+
+            // If an alternative repo owner was specified, update the intra-repo FROM commands.
+            bool updateDockerfile = !string.IsNullOrWhiteSpace(Options.RepoOwner)
+                && !image.ActivePlatform.FromImages.All(Manifest.IsExternalImage);
+            if (updateDockerfile)
+            {
+                string dockerfileContents = File.ReadAllText(dockerfilePath);
+
+                foreach (string fromImage in image.ActivePlatform.FromImages.SkipWhile(Manifest.IsExternalImage))
```
Comment (dagood) [inline src/Microsoft.DotNet.ImageBuilder/ImageBuilder.cs]: `SkipWhile` doesn't seem like what's intended here... I think you'll need `Where(x => !Manifest.IsExternalImage(x))`.

File: src/Microsoft.DotNet.ImageBuilder/Options.cs [hunk 3]
```diff
@@ -18,6 +18,7 @@ public class Options
 Usage:  image-builder [options]
 
 Options:
+      --repo-owner                      An alternative repo owner which overrides what is specified in the manifest
```
Comment (dagood) [inline src/Microsoft.DotNet.ImageBuilder/Options.cs]: Alphabetize?

File: src/Microsoft.DotNet.ImageBuilder/ImageBuilder.cs [hunk 4]
```diff
@@ -70,54 +72,26 @@ private static void Build()
...
             foreach (ImageInfo image in Manifest.ActiveImages)
             {
-                ExecuteHelper.Execute(
-                    "docker",
-                    $"build -t {string.Join(" -t ", image.ActiveFullyQualifiedTags)} {image.ActivePlatform.Model.Dockerfile}",
-                    Options.IsDryRun);
-            }
-        }
+                string dockerfilePath;
+                bool createdPrivateDockerfile = UpdateDockerfileFromCommands(image, out dockerfilePath);
```
Comment (MichaelSimons) [inline src/Microsoft.DotNet.ImageBuilder/ImageBuilder.cs]: I agree passing the Dockerfile as stdin would be ideal.  Unfortunately if you pass in the Dockerfile as stdin then there is no build context.  ASPNETCORE has files that must be included in the context therefore -f is the only option I can think of that would work.

File: src/Microsoft.DotNet.ImageBuilder/Options.cs (hunk 3 above)
Comment (MichaelSimons) [inline src/Microsoft.DotNet.ImageBuilder/Options.cs]: Yes - I renamed this and forgot the alphabetize - fixed.

File: src/Microsoft.DotNet.ImageBuilder/ImageBuilder.cs (hunk 2 above)
Comment (MichaelSimons) [inline src/Microsoft.DotNet.ImageBuilder/ImageBuilder.cs]: you are correct - changed

File: src/Microsoft.DotNet.ImageBuilder/ImageBuilder.cs (hunk 4 above)
Comment (dagood) [inline src/Microsoft.DotNet.ImageBuilder/ImageBuilder.cs]: Ah, unfortunate but it makes sense.

File: src/Microsoft.DotNet.ImageBuilder/ImageBuilder.cs [hunk 5]
```diff
@@ -191,6 +181,74 @@ private static void PushImages()
...
+                        filename = command.Substring(0, firstSpaceIndex);
+                        args = command.Substring(firstSpaceIndex + 1);
+                    }
+
+                    ExecuteHelper.Execute(filename, args, Options.IsDryRun);
+                }
+            }
+        }
+
+        private static bool UpdateDockerfileFromCommands(ImageInfo image, out string dockerfilePath)
```
Comment (dagood) [inline src/Microsoft.DotNet.ImageBuilder/ImageBuilder.cs]: The `bool` being the return value throws me off a little--makes it look like a `TryX` method but it's really not at all. Swapping them would help me, but I might have bad expectations. A factoring that doesn't use `out` would be appealing... I regret using it so much in VersionTools because it made some additions very awkward.

Also, isn't very clear to me what the return value means from the method name.

Comment (MichaelSimons) [inline src/Microsoft.DotNet.ImageBuilder/ImageBuilder.cs]: I apologize for my premature merge 😞.  I acknowledge the point you bring up and agree avoiding the `out` would be appealing.  IMO swapping the args doesn't address the problem.  I had thought about returning the dockerfile path only when re-written but that makes the calling code more complex in that it has to either re-calculate the path or conditionally specify the -f option.  Thinking...

Comment (dagood) [inline src/Microsoft.DotNet.ImageBuilder/ImageBuilder.cs]: Np, I think it's clear enough what it's doing to maintain it, just could be clearer.

Going a bit crazy with it, this could be solved by returning something `IDisposable` that provides a dockerfile path, and not even revealing to the caller whether or not it's a temp one.
